src/gencost/waterfall.py

```diff
@@ -1708,65 +1841,194 @@ def get_cems_by_x(self, subplant_id_col, xwalk=None, merge_only=False):
         if xwalk is None:
             xwalk = self.xwalk
 
-        merged = self.get_cems().merge(
-            xwalk[
-                [
-                    "plant_id_epa",
-                    "plant_id_eia",
-                    "emissions_unit_id_epa",
-                    subplant_id_col,
+        if subplant_id_col == "generator_id":
```

Comment:
I think the generator version of this is different enough that it could be it's own method without risking much if any duplication.